IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo

```diff
@@ -321,31 +416,40 @@ the simulation stops.");
       y=pCur3.dp);
   end if;
 
- // Compute derivatives for cubic spline
- motDer = if per.use_powerCharacteristic then zeros(size(per.motorEfficiency.V_flow,
-    1)) elseif (size(per.motorEfficiency.V_flow, 1) == 1) then {0} else
-    IBPSA.Utilities.Math.Functions.splineDerivatives(
-    x=per.motorEfficiency.V_flow,
-    y=per.motorEfficiency.eta,
-    ensureMonotonicity=IBPSA.Utilities.Math.Functions.isMonotonic(x=per.motorEfficiency.eta,
-      strict=false));
-  hydDer = if per.use_powerCharacteristic then zeros(size(per.hydraulicEfficiency.V_flow,
-    1)) elseif (size(per.hydraulicEfficiency.V_flow, 1) == 1) then {0}
-     else IBPSA.Utilities.Math.Functions.splineDerivatives(x=per.hydraulicEfficiency.V_flow,
-    y=per.hydraulicEfficiency.eta);
-
+  assert(not ((per.etaMotMet==
+           IBPSA.Fluid.Movers.BaseClasses.Types.MotorEfficiencyMethod.Efficiency_MotorPartLoadRatio
+           or  per.etaMotMet==
+           IBPSA.Fluid.Movers.BaseClasses.Types.MotorEfficiencyMethod.GenericCurve)
+         and not per.haveWMot_nominal),
+         "In " + getInstanceName() + ": etaMotMet is set to
+         .Efficiency_MotorPartLoadRatio or .GenericCurve which requires
+         the motor's rated power, but per.WMot_nominal is not assigned or
+         cannot be estimated because no power curve is provided.");
+
+  assert(max(per.power.P)<1E-6 or per.WMot_nominal>max(per.power.P)*0.99,
+         "In " + getInstanceName() + ": The rated motor power provided in
+         per.WMot_nominal is smaller than the maximum power provided in per.power.
+         Use a larger value for per.WMot_nominal or leave it blank to allow the
+         model to assume a default value.");
+
+initial algorithm
```

Comment:
Is it indeed required to put this assertion in an `initial algorithm` section as opposed to simply have it in `initial equation`?

Reply:
This particular assert warning no longer causes trouble with Dymola 2023x. This is now moved back to `initial equation` in e889c8e.
However, the other assert warnings in `PartialFlowMachine` still cause the simulation to stop incorrectly if they were moved from `initial algorithm` back to `initial equation`.